src/Containers.hs

```diff
@@ -229,7 +231,7 @@ class Container t => NontrivialContainer t where
   head = foldr (\x _ -> Just x) Nothing
   {-# INLINE head #-}
 
-instance {-# OVERLAPPABLE #-} Foldable f => NontrivialContainer (f a) where
+instance {-# OVERLAPPABLE #-} Foldable f => Container (f a) where
```

Comment:
One more thing: I would like to have this additional `newtype`

```haskell
newtype WrappedList l = WrappedList l
```

which is instance of `Container` using `toList` for every method: just delegates all methods to list implementation. This can be useful if you want to use `Container` methods for your data type but you don't want to implement 10 methods of this type class for you data type.